js/client-battle.js

```diff
@@ -22,6 +22,9 @@
 
 			this.$chat = this.$chatFrame.find('.inner');
 
+			this.$options = $('<div class="battle-options"></div>').insertAfter(this.$chatFrame.find('.inner-preempt')).html('<button name="toggleIgnoreSpects">' + (this.battle.ignoreSpects ? 'Unignore' : 'Ignore') + ' Spectators</button>');
+			this.battle.optionsElem = this.$options;
```

Comment:
According to the new Google thing we saw, we should initialize this in the constructor/prototype in `battle.js`.

And then we'd just do `this.$options = this.battle.optionsElem;` here.

Also prevents `Battle.prototype.log` from crashing in replays where `optionsElem` wouldn't otherwise exist.